### PR TITLE
[FEATURE] Added factory for items

### PR DESCRIPTION
**Pull request by turbo-ele**

### CODE REVIEW

File: Entity/BaseItem.php [hunk 1]
```diff
@@ -622,4 +632,51 @@ public function getTotalNetPrice()
...
+     * @param \DateTime $deliveryDate
+     *
+     * @return Item
+     */
+    public function setDeliveryDate($deliveryDate)
+    {
+        $this->deliveryDate = $deliveryDate;
+
+        return $this;
+    }
```
Comment (christianbader) [inline Entity/BaseItem.php:672]: add newline

File: Item/ItemFactory.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    private $defaultCurrencyCode;
+
+    /**
+     * @param string $apiProductEntity
+     */
+    public function __construct(
+        $apiProductEntity,
+        $defaultCurrencyCode
+    ) {
+        // FIXME: guess this is not the way this should be done
```
Comment (christianbader) [inline Item/ItemFactory.php:28]: bam!

**Comment by christianbader** [comment]
ack

File: Item/ItemFactory.php (hunk 2 above)
Comment (turbo-ele) [inline Item/ItemFactory.php:28]: will be replaced by productfactory

**Pull request merged**